tests/Controller/ProductReviewApiTest.php

```diff
@@ -0,0 +1,360 @@
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Review\Model\ReviewInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @author Paul Stoica <paul.stoica18@gmail.com>
+ */
+final class ProductReviewApiTest extends JsonApiTestCase
+{
+
+    /**
+     * @var array
+     */
+    private static $authorizedHeaderWithContentType = [
+        'HTTP_Authorization' => 'Bearer SampleTokenNjZkNjY2MDEwMTAzMDkxMGE0OTlhYzU3NzYyMTE0ZGQ3ODcyMDAwM2EwMDZjNDI5NDlhMDdlMQ',
+        'CONTENT_TYPE'       => 'application/json',
+    ];
+
+    /**
+     * @var array
+     */
+    private static $authorizedHeaderWithAccept = [
+        'HTTP_Authorization' => 'Bearer SampleTokenNjZkNjY2MDEwMTAzMDkxMGE0OTlhYzU3NzYyMTE0ZGQ3ODcyMDAwM2EwMDZjNDI5NDlhMDdlMQ',
+        'ACCEPT'             => 'application/json',
+    ];
+
+    /**
+     * @test
+     */
+    public function it_does_not_allows_showing_product_review_list_when_access_is_denied()
+    {
+        $productReviewsData = $this->loadFixturesFromFile('resources/product_reviews.yml');
+
+        /** @var ProductInterface $product */
+        $product = $productReviewsData['product1'];
+
+        $this->client->request('GET', $this->getReviewListUrl($product));
+        $response = $this->client->getResponse();
+
+        $this->assertResponse($response, 'authentication/access_denied_response', Response::HTTP_UNAUTHORIZED);
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_not_allows_showing_product_review_when_it_does_not_exist()
+    {
+        $productReviewsData = $this->loadFixturesFromFile('resources/product_reviews.yml');
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+
+        /** @var ProductInterface $product */
+        $product = $productReviewsData['product1'];
+
+        $this->client->request('GET', $this->getReviewListUrl($product) . '0', [], [], static::$authorizedHeaderWithAccept);
+        $response = $this->client->getResponse();
+
+        $this->assertResponse($response, 'error/not_found_response', Response::HTTP_NOT_FOUND);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_showing_product_review()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $productReviewsData = $this->loadFixturesFromFile('resources/product_reviews.yml');
+
+        /** @var ProductInterface $product */
+        $product = $productReviewsData['product1'];
+
+        /** @var ReviewInterface $productReview */
+        $productReview = $productReviewsData['productReview1'];
+
+        $this->client->request('GET', $this->getReviewUrl($product, $productReview), [], [], static::$authorizedHeaderWithAccept);
+        $response = $this->client->getResponse();
+
+        $this->assertResponse($response, 'product_review/show_response', Response::HTTP_OK);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_indexing_product_reviews()
```

Comment:
Missing denied access test

Reply:
@lchrusciel  there is a test method: "it_does_not_allows_showing_product_review_list_when_access_is_denied", or what do you mean?

Reply:
I just missed it